### PR TITLE
[feat] FastVideo World Model Training

### PR DESCRIPTION
**Pull request by H1yori233**
## Purpose

Adds training recipes for the **MatrixGame 2.0** I2V world model on both the legacy and the new trainer.

## Changes

**Legacy framework**:
- `matrixgame_ar_diffusion_pipeline.py` — AR diffusion-forcing
- `matrixgame_ode_causal_pipeline.py` — ODE-trajectory init
- `matrixgame_self_forcing_distillation_pipeline.py` — Self-Forcing DMD distillation

**New framework**:
- `fastvideo/train/models/matrixgame/{matrixgame.py, matrixgame_causal.py}` — bidirectional + causal student plugins
- `examples/train/scenario/worldmodel/` — `finetune_i2v.yaml`, `dfsft_causal_i2v.yaml`, `self_forcing_causal_i2v.yaml`, `README.md`

## Checklist

- [X] I ran `pre-commit run --all-files` and fixed all issues

### CODE REVIEW

File: fastvideo/dataset/dataloader/record_schema.py [hunk 1]
```diff
@@ -188,3 +188,110 @@ def text_only_record_creator(text_name: str, text_embedding: np.ndarray,
...
+        "trajectory_latents_bytes": trajectory_latents.tobytes(),
+        "trajectory_latents_shape": list(trajectory_latents.shape),
+        "trajectory_latents_dtype": str(trajectory_latents.dtype),
+    })
+
+    record.update({
+        "trajectory_timesteps_bytes": trajectory_timesteps.tobytes(),
+        "trajectory_timesteps_shape": list(trajectory_timesteps.shape),
+        "trajectory_timesteps_dtype": str(trajectory_timesteps.dtype),
+    })
```
[the remark this answers is not in the document]
Comment (H1yori233) [reply fastvideo/dataset/dataloader/record_schema.py:298]: I think there is no need to do this?

File: fastvideo/training/training_pipeline.py
Comment (mignonjia) [inline fastvideo/training/training_pipeline.py:1]: How about just use global_step == self.init_steps to check ref video logged or not, instead of self.validation_ref_videos_logged = False at init.

File: fastvideo/models/utils.py
Comment (mignonjia) [inline fastvideo/models/utils.py:1]: Why is sigma_from_timestep needed? Seems timesteps are all precomputed, and the sigmas are stored in scheduler.sigmas based on self.timesteps.

File: fastvideo/pipelines/preprocess/matrixgame/matrixgame_preprocess_pipeline.py [hunk 2]
```diff
@@ -60,7 +60,7 @@ def get_extra_features(self, valid_data: dict[str, Any], fastvideo_args: FastVid
             clip_features = clip_features.last_hidden_state
 
         features["clip_feature"] = clip_features
-        """Get VAE features from the first frame of each video"""
+        # Get CLIP features from the first frame of each video.
```
[the remark this answers is not in the document]
Comment (mignonjia) [reply fastvideo/pipelines/preprocess/matrixgame/matrixgame_preprocess_pipeline.py:63]: Yes this should be VAE, you can fix it

File: examples/train/configs/fine_tuning/wan/dfsft_causal_t2v.yaml
Comment (mignonjia) [inline examples/train/configs/fine_tuning/wan/dfsft_causal_t2v.yaml:1]: what is shift 5 gaussian weight mean here?

**Review by mignonjia — APPROVED** [review]
